client/desktop/lib/cookie-auth/index.js

```diff
@@ -70,49 +67,62 @@ function parseCookie( cookieStr ) {
 	}, cookie );
 }
 
-function setSessionCookies( window, onComplete ) {
-	return function ( response ) {
-		let cookieHeaders = response.headers[ 'set-cookie' ];
+function setSessionCookies( window, authorizeResponse ) {
+	return new Promise( ( resolve ) => {
+		let cookieHeaders = authorizeResponse.headers[ 'set-cookie' ];
 		let count = 0;
 		if ( ! Array.isArray( cookieHeaders ) ) {
 			cookieHeaders = [ cookieHeaders ];
 		}
```

Comment:
I'm wondering what this code does when there are no 'set-cookies' headers in the response. Then `cookieHeaders` are `undefined`, are converted to `[ undefined ]` and then `parseCookie` crashed when trying to call `.split()` on `undefined` ? I didn't test it, but I suspect there is a pre-existing bug.